### DISCUSSION TITLE
I have highlighting errors in my VSCode when working with solidity

### DISCUSSION THREAD

**Discussion by iamnotyou3** [Cyfrin/foundry-full-course-cu#1644] [Q&A]
Hi there!

I have installed "solidity" extension by Juan Blanko to my vscode to highlight solidity correctly but seems I haven't configured everything correctly. 

1. i have an error 
`"Source "forge-std/Script.sol" not found: File import callback not supported"`

on the line

`import {Script} from "forge-std/Script.sol";`

but everything compiles fine

2. `foundry.toml` file is not highlighted in any way, but as I have found it should be after installing "solidity"

3. I added a parameter to my contract constructor, but when I call constructor without parameter, I don't have it highlighted as error

I haven't found any relevant info, has anyone faced the same issue?

**Comment by EngrPips** [comment]
Hello @iamnotyou3

1. You have to install `forge-std` using the command `forge install foundry-rs/forge-std --no-commit` to be able to make an import from the `forge-std` library.
2. You can install the `Solidity Visual Developer` extension and see if that would give your `foundry.toml` file some highlighting.
3. Sometimes, `vscode` takes time to load fully, and it sometimes glitches and does not run the file correctly for errors. I will advise close and reopen your `vscode`

**Reply by iamnotyou3** [reply]
Hi @EngrPips, thanks for your answer! Actually `forge-std` is already installed, but looks like vscode and foundry are not "friends" in my case, so extensions does not understand that `"forge-std/Script.sol"` is actually `"lib/forge-std/Script.sol"` according to `foundry.toml`

**Reply by EngrPips** [reply]
Oh, I see. Did you successfully import it now?

**Reply by serjou1** [reply]
i can easily compile it with forge, but vscode highlightes it as an error ignoring foundry.toml remappings

**Reply by EngrPips** [reply]
Maybe close and reopen your `vscode`, and since the code compile successfully, I believe the vscode highlight is incorrect and can be ignored.

**Reply by serjou1** [reply]
Yes, it is incorrect, but I would like it to be correct :)
I guess I should write my own extension for this...

**Reply by EngrPips** [reply]
Yeah, why not? Go for it!

**Reply by FrankezeCode** [reply]
> i can easily compile it with forge, but vscode highlightes it as an error ignoring foundry.toml remappings

Please how did you resolve this highlighting issue, I'm facing the same issue

**Reply by AbigirlMuchineripi** [reply]
hie @FrankezeCode  check if you only have forge std installed in your lib folder, if so, run forge install smartcontractkit/chainlink-brownie-contracts to install the whole chainlink library, change your import to..lib/chainlink-brownie-contracts/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol and also change your input in PriceConverter.sol .

If this worked mark it as an answer lol.

**Reply by FrankezeCode** [reply]
Thanks I have resolved the issue 🙏,   I'm not the creator of the the discussion , I'm unable to mark it as answered

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.

**Comment by AbigirlMuchineripi** [comment]
Hello, maybe you should try the Solidity Nomic Foundation extension and make use of Github Copilot, as for me I am using Tabnine Pro in Vs code, you can use it for free for 3 months. It's been helpful.

**Reply by EngrPips** [reply]
Great suggestion here @serjou1.
